parquet-thrift/src/main/java/parquet/thrift/BufferedProtocolReadToWrite.java

```diff
@@ -381,6 +389,8 @@ public String toDebugString() {
             // read path.
             throw new ParquetRuntimeException("This should never happen! "
                 + "Don't know if this field is a union, was the deprecated constructor of StructType used?\n" + type){};
+          default:
+            throw new ParquetRuntimeException("Unrecognized StructOrUnionType: " + type.getStructOrUnionType()){};
```

Comment:
we should avoid those anonymous subclasses and just create a concrete one.
Something like ShouldNotHappenException?